chain/indexer/src/streamer/senders.rs

```diff
@@ -0,0 +1,207 @@
+use std::collections::HashMap;
+
+use futures::StreamExt;
+use near_async::messaging::{AsyncSender, SendAsync};
+use near_chain_configs::ProtocolConfigView;
+use near_client::{
+    GetBlock, GetChunk, GetExecutionOutcomesForBlock, GetProtocolConfig, GetReceipt,
+    GetStateChangesWithCauseInBlockForTrackedShards, Status, StatusResponse,
+};
+use near_client_primitives::types::{
+    GetBlockError, GetChunkError, GetProtocolConfigError, GetReceiptError, GetStateChangesError,
+    StatusError,
+};
+use near_epoch_manager::shard_tracker::ShardTracker;
+use near_indexer_primitives::IndexerExecutionOutcomeWithOptionalReceipt;
+use near_o11y::span_wrapped_msg::{SpanWrapped, SpanWrappedMessageExt};
+use near_primitives::hash::CryptoHash;
+use near_primitives::types::{BlockId, BlockReference, ShardId};
+use near_primitives::views::{
+    BlockView, ChunkView, ExecutionOutcomeWithIdView, ReceiptView, StateChangesView,
+};
+use tracing::warn;
+
+use crate::INDEXER;
+use crate::streamer::errors::FailedToFetchData;
+
+#[derive(Clone, near_async::MultiSend, near_async::MultiSenderFrom)]
+pub struct IndexerViewClientSender {
+    pub get_block_sender: AsyncSender<GetBlock, Result<BlockView, GetBlockError>>,
+    pub get_chunk_sender: AsyncSender<GetChunk, Result<ChunkView, GetChunkError>>,
+    pub get_protocol_config_sender:
+        AsyncSender<GetProtocolConfig, Result<ProtocolConfigView, GetProtocolConfigError>>,
+    pub get_execution_outcomes_for_block_sender: AsyncSender<
+        GetExecutionOutcomesForBlock,
+        Result<HashMap<ShardId, Vec<ExecutionOutcomeWithIdView>>, String>,
+    >,
+    pub get_receipt_sender: AsyncSender<GetReceipt, Result<Option<ReceiptView>, GetReceiptError>>,
+    pub get_state_changes_with_cause_in_block_for_tracked_shards_sender: AsyncSender<
+        GetStateChangesWithCauseInBlockForTrackedShards,
+        Result<HashMap<ShardId, StateChangesView>, GetStateChangesError>,
+    >,
+}
+
+#[derive(Clone, near_async::MultiSend, near_async::MultiSenderFrom)]
+pub struct IndexerClientSender {
+    pub status_sender: AsyncSender<SpanWrapped<Status>, Result<StatusResponse, StatusError>>,
+}
+
+impl IndexerViewClientSender {
+    pub(crate) async fn fetch_block(
+        &self,
+        hash: CryptoHash,
+    ) -> Result<BlockView, FailedToFetchData> {
+        tracing::debug!(target: INDEXER, ?hash, "fetch block by hash");
+        self.send_async(GetBlock(BlockId::Hash(hash).into()))
+            .await?
+            .map_err(|err| FailedToFetchData::String(err.to_string()))
+    }
+
+    pub(crate) async fn fetch_latest_block(
+        &self,
+        finality: near_primitives::types::Finality,
+    ) -> Result<BlockView, FailedToFetchData> {
+        tracing::debug!(target: INDEXER, ?finality, "fetch latest block");
+        self.send_async(near_client::GetBlock(near_primitives::types::BlockReference::Finality(
```

Comment:
nit: Could we import the types here? Example, use `GetBlock` instead of `near_client::GetBlock`, which I think is already imported. There are several places in the file, `BlockReference`, `BlockId`, `FuturesUnordered`, `ProtocolConfigView`, `CryptoHash`, etc. etc.

Reply:
yes, definitely, thanks for pointing out: 12c8ed7b1